### PR TITLE
Okina wip

### PR DESCRIPTION
**Pull request by camierjs**
- Cleanup
- ex1 and ex1p work in FA or PA, on CPU or GPU and w/ or w/o MPI
```
make config MFEM_CXX="nvcc" \
           CXXFLAGS="-O3 --restrict --expt-extended-lambda -x=cu -arch=sm_60 -std=c++11 -m64 -Xcompiler -Wall" \
           MFEM_USE_MPI=YES HYPRE_DIR=~/usr/local/hypre/2.11.2 MFEM_USE_METIS_5=YES METIS_DIR=~/usr/local/metis/5.1.0 \
           MFEM_TPLFLAGS="-I/home/camier1/usr/local/hypre/2.11.2/include \
                          -I/home/camier1/usr/local/metis/5.1.0/include \
                          -I/home/camier1/usr/local/openmpi/3.0.0/include" \
           MFEM_EXT_LIBS="-lrt -lcuda -lcudart -lcudadevrt -lnvToolsExt \
                          -L/home/camier1/usr/local/hypre/2.11.2/lib -lHYPRE \
                          -L/home/camier1/usr/local/metis/5.1.0/lib -lmetis \
                          -L/home/camier1/usr/local/openmpi/3.0.0/lib -lmpi"
```

### CODE REVIEW

File: linalg/kernels/vector.hpp [hunk 1]
```diff
@@ -29,6 +29,26 @@ void kVectorAlphaAdd(double *vp, const double* v1p,
 
 void kVectorPrint(const size_t N, const double *data);
 
+// *****************************************************************************
+template<typename T>
+void kVectorSetK(T *data, const size_t k, T value)
+{
+   GET_CUDA;
+   GET_ADRS_T(data,T);
+   if (cuda) { assert(false); }
```
[the remark this answers is not in the document]
Comment (camierjs) [reply linalg/kernels/vector.hpp]: Right, I should have. 
I can even remove these Set/Get as they are not used anymore.

File: general/kernels.hpp [hunk 2]
```diff
@@ -1,50 +0,0 @@
-// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
```
[the remark this answers is not in the document]
Comment (camierjs) [reply general/kernels.hpp]: I was going for `okcuda.?pp`, `okocca.?pp` & so on, `kernels` seems too generic.
I'm open to other propositions, sure.

File: general/custub.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
```
[the remark this answers is not in the document]
Comment (camierjs) [reply general/custub.cpp]: Yes, `okcuda.?pp` or just `cuda.?pp`

File: general/kmalloc.hpp [hunk 4]
```diff
@@ -1,85 +0,0 @@
-// Copyright (c) 2017, Lawrence Livermore National Security, LLC. Produced at
-// the Lawrence Livermore National Laboratory. LLNL-CODE-734707. All Rights
-// reserved. See files LICENSE and NOTICE for details.
-//
-// This file is part of CEED, a collection of benchmarks, miniapps, software
```
[the remark this answers is not in the document]
Comment (camierjs) [reply general/kmalloc.hpp]: This file was deleted, but I'll go through the headers.

File: general/custub.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (tzanio) [reply general/custub.cpp]: Naming is hard... lets discuss in person :)

**Pull request merged**